File: go-app/v2_traduccion.go
package main

import (
	"bytes"
	"encoding/json"
	"encoding/xml"
	"io"
	"net/http"
	"net/url"
	"strings"
	"github.com/gofiber/fiber/v2"
)

type SoapResponseV2 struct {
	XMLName xml.Name `xml:"Envelope"`
	Body    struct {
		NumberToWordsResponse struct {
			NumberToWordsResult string `xml:"NumberToWordsResult"`
		} `xml:"NumberToWordsResponse"`
	} `xml:"Body"`
}

// Estructura para decodificar la respuesta JSON del traductor MyMemory
type MyMemoryResponse struct {
	ResponseData struct {
		TranslatedText string `json:"translatedText"`
	} `json:"responseData"`
}

func init() {
	RegistrarV2Traduccion = func(app *fiber.App) {
		// Ruta exacta: http://127.0.0.1:8000/clisoap2?n=10
		app.Get("/clisoap2", func(c *fiber.Ctx) error {
			numero := c.Query("n", "0")

			// 1. Consumir el SOAP (Misma lógica real de la V1)
			sobreSoap := `<?xml version="1.0" encoding="utf-8"?>
			<soap:Envelope xmlns:soap="http://schemas.xmlsoap.org/soap/envelope/">
			  <soap:Body>
				<NumberToWords xmlns="http://www.dataaccess.com/webservicesserver/">
				  <ubiNum>` + numero + `</ubiNum>
				</NumberToWords>
			  </soap:Body>
			</soap:Envelope>`

			respSoap, err := http.Post("https://www.dataaccess.com/webservicesserver/NumberConversion.wso", "text/xml; charset=utf-8", bytes.NewBufferString(sobreSoap))
			if err != nil {
				return c.Status(500).SendString("Error en SOAP")
			}
			defer respSoap.Body.Close()

			bodySoap, _ := io.ReadAll(respSoap.Body)
			var resultadoSoap SoapResponseV2
			xml.Unmarshal(bodySoap, &resultadoSoap)
			textoIngles := strings.TrimSpace(resultadoSoap.Body.NumberToWordsResponse.NumberToWordsResult)

			// 2. Consumir la API real de MyMemory para traducir de inglés a español
			urlTraductor := "https://api.mymemory.translated.net/get?q=" + url.QueryEscape(textoIngles) + "&langpair=en|es"
			respTrad, err := http.Get(urlTraductor)
			if err != nil {
				return c.Status(500).SendString("Error en la traducción")
			}
			defer respTrad.Body.Close()

			bodyTrad, _ := io.ReadAll(respTrad.Body)
			var resultadoTrad MyMemoryResponse
			json.Unmarshal(bodyTrad, &resultadoTrad)

			textoEspanol := strings.ToLower(resultadoTrad.ResponseData.TranslatedText)
			c.Set("Content-Type", "text/plain; charset=utf-8")
			return c.SendString(textoEspanol)
		})
	}
}

File: dotnet-app/DotnetApp/v2_traduccion.cs
using Microsoft.AspNetCore.Builder;
using Microsoft.AspNetCore.Http;
using System;
using System.Linq;
using System.Net.Http;
using System.Text;
using System.Text.Json;
using System.Xml.Linq;

public static class V2Traduccion
{
    public static void RegistrarRuta(WebApplication app)
    {
        // Ruta exacta requerida: http://127.0.0.1:8000/clisoap2?n=10
        app.MapGet("/clisoap2", async (HttpContext context) =>
        {
            string numero = context.Request.Query["n"].ToString() ?? "0";

            // 1. Consumir el servicio SOAP (Misma lógica real de la V1)
            string sobreSoap = $@"<?xml version=""1.0"" encoding=""utf-8""?>
            <soap:Envelope xmlns:soap=""http://schemas.xmlsoap.org/soap/envelope/"">
              <soap:Body>
                <NumberToWords xmlns=""http://www.dataaccess.com/webservicesserver/"">
                  <ubiNum>{numero}</ubiNum>
                </NumberToWords>
              </soap:Body>
            </soap:Envelope>";

            using var cliente = new HttpClient();
            var contenido = new StringContent(sobreSoap, Encoding.UTF8, "text/xml");
            var respuestaSoap = await cliente.PostAsync("https://www.dataaccess.com/webservicesserver/NumberConversion.wso", contenido);
            
            if (!respuestaSoap.IsSuccessStatusCode)
                return Results.Text("Error en SOAP", "text/plain");

            string xmlResultado = await respuestaSoap.Content.ReadAsStringAsync();
            var documento = XDocument.Parse(xmlResultado);
            XNamespace ns = "http://www.dataaccess.com/webservicesserver/";
            string textoIngles = documento.Descendants(ns + "NumberToWordsResult").FirstOrDefault()?.Value?.Trim() ?? "";

            // 2. Consumir la API real de MyMemory para traducir el resultado al vuelo
            string urlTraductor = $"https://api.mymemory.translated.net/get?q={Uri.EscapeDataString(textoIngles)}&langpair=en|es";
            var respuestaTraduccion = await cliente.GetAsync(urlTraductor);
            
            if (respuestaTraduccion.IsSuccessStatusCode)
            {
                string jsonResultado = await respuestaTraduccion.Content.ReadAsStringAsync();
                using var jsonDoc = JsonDocument.Parse(jsonResultado);
                string textoEspanol = jsonDoc.RootElement.GetProperty("responseData").GetProperty("translatedText").GetString() ?? "Error";

                return Results.Text(textoEspanol.ToLower(), "text/plain; charset=utf-8");
            }

            return Results.Text("Error en la traducción", "text/plain");
        });
    }
}